packages/typespec-vscode/src/vscode-cmd/create-tsp-project.ts

```diff
@@ -300,146 +325,114 @@ async function selectEmitters(
   return Object.fromEntries(selectedEmitters.map((x) => [x.name, x.emitterTemplate]));
 }
 
-async function tspInstall(
-  client: TspLanguageClient,
-  directory: string,
-): Promise<ExecOutput | undefined> {
-  logger.info("Installing TypeSpec project dependencies by 'tsp install'...");
-  return await vscode.window.withProgress(
-    {
-      location: vscode.ProgressLocation.Notification,
-      title: "Installing TypeSpec project dependencies by 'tsp install'...",
-      cancellable: true,
-    },
-    async (_progress, token) => {
-      const TIMEOUT = 600000; // set timeout to 10 minutes which should be enough for tsp install for a new project
-      try {
-        const result = await createPromiseWithCancelAndTimeout(
-          client.runCliCommand(["install"], directory),
-          token,
-          TIMEOUT,
-        );
-        return result;
-      } catch (e) {
-        if (e === ResultCode.Cancelled) {
-          logger.info(
-            "Installation of TypeSpec project dependencies by 'tsp install' is cancelled by user",
-          );
-          return undefined;
-        } else if (e === ResultCode.Timeout) {
-          logger.error(
-            `Installation of TypeSpec project dependencies by 'tsp install' is timeout after ${TIMEOUT}ms`,
-          );
-          return undefined;
+async function installDependencies(directory: string): Promise<Result<ExecOutput | undefined>> {
+  logger.info("Installing project dependencies");
+  const checkTspCliPromise = checkInstalledTspCli();
+  const checkNpmPromise = checkInstalledNpm();
+  const [checkTspCli, checkNpm] = await Promise.all([checkTspCliPromise, checkNpmPromise]);
+  if (checkTspCli || checkNpm) {
+    return await tryExecuteWithUi(
+      {
+        name: "Installing project dependencies",
+        progress: {
+          timeoutInMs: 5 * 60 * 1000, // 5 minutes
+          title: "Installing project dependencies",
+          withCancelAndTimeout: true,
+        },
+      },
+      async (progress) => {
+        if (checkTspCli) {
+          logger.info("tsp cli is installed, try to install dependencies with tsp install");
+          progress?.report({ message: "running 'tsp install' ..." });
+          return await spawnExecutionAndLogToOutput("tsp", ["install"], directory);
```

Comment:
to be inline with the init of the compiler should maybe run with `--save-package-manager` option but only when doing the init

Reply:
updated

Reply:
reverted to not include --save-package-manager because it will be complained by the old compiler which doesn't support the option. We can add it back later when most people upgrade to the compiler supporting the option. Or if we can bundle the logic directly, then this won't be a problem anymore. thx.